demos/http/http_demo_plaintext/http_demo_plaintext.c

```diff
@@ -0,0 +1,524 @@
+    }
+    else
+    {
+        LogError( ( "Failed to send HTTP %s request to %s%s: Error=%s.",
+                    pMethod, SERVER_HOST, pPath, HTTPClient_strerror( httpStatus ) ) );
+    }
+
+    if( httpStatus == HTTP_SUCCESS )
+    {
+        return EXIT_SUCCESS;
+    }
+    else
+    {
+        return EXIT_FAILURE;
+    }
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Entry point of demo.
+ *
+ * This example resolves a domain, then establishes a plaintext TCP connection to
+ * an HTTP server. After which, HTTP_Client APIs are used to send a GET, HEAD,
+ * PUT, and POST request in that order. For each request, an HTTP response from
+ * the server (or an error code) is logged. Note that this example is
+ * single-threaded and uses statically allocated memory.
+ *
+ */
+int main( int argc,
+          char ** argv )
+{
+    int returnStatus = EXIT_SUCCESS;
+
+    ( void ) argc;
+    ( void ) argv;
+
+    /**************************** Connect. ******************************/
+
+    /* Establish TCP connection. */
+    returnStatus = connectToServer( SERVER_HOST, SERVER_PORT, &socketContext.tcpSocket );
+
+    /* Define the transport interface. */
+    if( returnStatus == EXIT_SUCCESS )
+    {
+        transportInterface.recv = transportRecv;
+        transportInterface.send = transportSend;
+        transportInterface.pContext = &socketContext;
+    }
+
+    /*********************** Send HTTPS request. ************************/
+
+    /* Send GET Request. */
+    if( returnStatus == EXIT_SUCCESS )
+    {
+        returnStatus = sendHttpRequest( &transportInterface,
+                                        SERVER_HOST,
+                                        HTTP_METHOD_GET,
+                                        GET_PATH );
+    }
+
+    /* Send HEAD Request. */
+    if( returnStatus == EXIT_SUCCESS )
+    {
+        returnStatus = sendHttpRequest( &transportInterface,
+                                        SERVER_HOST,
+                                        HTTP_METHOD_HEAD,
+                                        HEAD_PATH );
+    }
+
+    /* Send PUT Request. */
+    if( returnStatus == EXIT_SUCCESS )
+    {
+        returnStatus = sendHttpRequest( &transportInterface,
+                                        SERVER_HOST,
+                                        HTTP_METHOD_PUT,
+                                        PUT_PATH );
+    }
+
+    /* Send POST Request. */
+    if( returnStatus != EXIT_SUCCESS )
+    {
+        returnStatus = sendHttpRequest( &transportInterface,
+                                        SERVER_HOST,
+                                        HTTP_METHOD_POST,
+                                        POST_PATH );
+    }
+
+    /************************** Disconnect. *****************************/
+
+    if( socketContext.tcpSocket != -1 )
+    {
+        shutdown( socketContext.tcpSocket, SHUT_RDWR );
+        close( socketContext.tcpSocket );
```

Comment:
Should these be `( void )`'ed since they return `int`s? I don't believe we're running MISRA on demos though so it should be fine